clients/src/main/resources/common/message/AlterPartitionRequest.json

```diff
@@ -18,16 +18,21 @@
   "type": "request",
   "listeners": ["zkBroker", "controller"],
   "name": "AlterPartitionRequest",
-  "validVersions": "0-1",
+  // Version 1 adds LeaderRecoveryState field (KIP-704).
+  //
+  // Version 2 adds TopicId field to replace TopicName field (KIP-841).
+  "validVersions": "0-2",
```

Comment:
> if topic id is enabled (IBP 2.8) is it guarantee that every topic will have an id

No, it is not. The broker could run with IBP 2.8 but have not topic ids yet because the controller does not use IBP 2.8 yet. In the end, we can use version 2 only if all the pending changes have a topic id. We have to use version 1 otherwise. Let me add a comment about this where we build the request.